setup.py

```diff
@@ -45,7 +45,7 @@
     extras_require={
         "dev": ["black", "isort", "pytest<8.0", "flake8", "pre-commit"],
         "atari": ["gym[atari, accept-rom-license]"],
-        "mujoco": ["gym[mujoco, mujoco_py]"],
+        "mujoco": ["gym[mujoco]", "mujoco", "mujoco_py>=2.0,<3.0"],
```

Comment:
Why do we need to set the version? I think gym will automatically import the latest mujoco.

Reply:
The 2nd mujoco and mujoco_py are not from gym and need to be installed seperately. The versioning on mujoco_py is probably not necessary but I included it anyways

Reply:
I wonder why tests worked without it :)

Btw I think restricting the major version of a package is a good practice @wmFrank 
The reason is that major version updates usually introduce incompatible API changes. Unless we want our software to break every time authors of another package release a new major version, we should use limits like this.

One example is going to be sample factory 2.0. If someone has just sample_factory in their dependencies, us releasing a new PyPI package is going to break their library.